searchapp/generate_doc.py

```diff
@@ -2,272 +2,70 @@
-                p1=document.add_paragraph(str(i)+": ")
-                p1.add_run(question)
-    
-    para = document.add_paragraph()
-    run = para.add_run()
-    run.add_break()
-    ct = 0
-    for row in questions_mapping:
-        if row.get('question_type')=='2':
-            for question in row['question']:
-                ct = ct+1
-    section=document.add_paragraph()
-    section.add_run("Section C").bold=True
-    section.alignment = 1
-    subsection = document.add_paragraph()
-    subsection.add_run(str(ct) + " questions of 2 marks each")
-    subsection.alignment = 1
-    for row in questions_mapping:
-        if row['question_type'] == '2':
-            p=document.add_paragraph()
-            p.add_run("Attempt only "+str(min(row['attempt'],ct))+" questions").italic=True
-            p.alignment = 1
-            for question in row['question']:
-                i=i+1
-                p1=document.add_paragraph(str(i)+": ")
-                p1.add_run(question)
-    
-    para = document.add_paragraph()
-    run = para.add_run()
-    run.add_break()
-    ct = 0
-    for row in questions_mapping:
-        if row.get('question_type')=='3':
-            for question in row['question']:
-                ct = ct+1
-    section=document.add_paragraph()
-    section.add_run("Section D").bold=True
-    section.alignment = 1
-    subsection = document.add_paragraph()
-    subsection.add_run(str(ct) + " questions of 3 marks each")
-    subsection.alignment = 1
-    for row in questions_mapping:
-        if row['question_type'] == '3':
-            p=document.add_paragraph()
-            p.add_run("Attempt only "+str(min(row['attempt'],ct))+" questions").italic=True
-            p.alignment = 1
-            for question in row['question']:
-                i = i+1
-                p1 = document.add_paragraph(str(i)+" : ")
-                p1.add_run(question)
+    return question_no
 
-    para = document.add_paragraph()
-    run = para.add_run()
-    run.add_break()
-    ct = 0
-    for row in questions_mapping:
-        if row.get('question_type')=='5':
-            for question in row['question']:
-                ct = ct+1
-    section=document.add_paragraph()
-    section.add_run("Section E").bold=True
-    section.alignment = 1
-    subsection = document.add_paragraph()
-    subsection.add_run(str(ct) + " questions of 5 marks each")
-    subsection.alignment = 1
-    for row in questions_mapping:
-        if row['question_type'] == '5':
-            p = document.add_paragraph()
-            p.add_run("Attempt only "+str(min(row['attempt'],ct))+" questions").italic=True
-            p.alignment = 1
-            for question in row['question']:
-                i = i+1
-                p1 = document.add_paragraph(str(i)+" : ")
-                p1.add_run(question)
+def convert_customized_to_doc(questions_mapping, subject):
+    document = Document()
+    for name in questions_mapping:
+        pp = document.add_paragraph()
```

Comment:
Well since we are refactoring all this anyways, lets get away with using variables like `pp` or `stud_name`. Let's be as verbose as possible while writing code